example/gauss_laguerre_quadrature.cpp

```diff
@@ -225,7 +225,7 @@ class guass_laguerre_abscissas_and_weights : private boost::noncopyable
         // bisection steps in order to tighten up the root bracket.
         boost::uintmax_t a_couple_of_iterations = 3U;
         const std::pair<T, T>
-          first_laguerre_root = boost::math::tools::bisect(laguerre_object,
+          first_laguerre_roots = boost::math::tools::bisect(laguerre_object,
```

Comment:
@NAThompson : `first_laguerre_root`(s) appears to be unused?

There also seem to be some branches in the logic which do not set `first_laguerre_root_has_been_found`?

Reply:
I'm baffled! I'm pretty sure I didn't write any of this, but there's my name in the copyright! I read the `git blame`; looks like @ckormanyos write it.

Reply:
I'll try to establish the true author, and absolve you of all possible blame for any hanging branches!

Reply:
> looks like @ckormanyos ... [wrote] it

I did. I will look at the example and see if the variable is used or not.
I will also check that example.

Reply:
>>> There also seem to be some branches in the logic which do not set `first_laguerre_root_has_been_found`?

Correct. Thanks. Good catch. I have refactored these issues.

>>I'll try to establish the true author

> I will look at the example and see if the variable is used or not.

I have refactored the example Gauss Laguerre quadrature
[here](http://github.com/boostorg/multiprecision/tree/gauss_laguerre)

Reply:
See #196
We can merge those changes to this PR if desired.